modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseUpgradeCheck.java

```diff
@@ -46,13 +54,85 @@ protected String addNewImports(String newContent) {
 		return newContent;
 	}
 
+	protected String addNewImportsJSP(String newContent, String... newImports) {
+		if (newImports.length == 0) {
+			return newContent;
+		}
+
+		Set<String> missingImports = new TreeSet<>();
+
+		Collections.addAll(missingImports, newImports);
+
+		if (missingImports.isEmpty()) {
+			return newContent;
+		}
+
```

Comment:
@TamyrisTorres this `if` isn't unnecessary?

Reply:
resolved.